### PR TITLE
Add user model and authentication.

### PR DESCRIPTION
**Pull request by bin0mial**
## The Changes Include:
- Adding User Model
- Adding User Migration
- Authentication using Devise gem (Login, Register)

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -28,9 +28,13 @@ gem 'jbuilder', '~> 2.7'
 # Reduces boot times through caching; required in config/boot.rb
 gem 'bootsnap', '>= 1.4.4', require: false
 
+gem 'devise'
+
 group :development, :test do
   # Call 'byebug' anywhere in the code to stop execution and get a debugger console
   gem 'byebug', platforms: [:mri, :mingw, :x64_mingw]
+  gem 'factory_bot_rails'
+  gem 'faker'
```
Comment (a-ali) [inline Gemfile]: Please specify the gems versions (Check the Gemfile.lock), .i.e.

```ruby

gem 'devise', '4.8.0'`

....
```

File: config/locales/devise.en.yml [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  errors:
+    messages:
+      already_confirmed: "was already confirmed, please try signing in"
+      confirmation_period_expired: "needs to be confirmed within %{period}, please request a new one"
+      expired: "has expired, please request a new one"
+      not_found: "not found"
+      not_locked: "was not locked"
+      not_saved:
+        one: "1 error prohibited this %{resource} from being saved:"
+        other: "%{count} errors prohibited this %{resource} from being saved:"
```
Comment (a-ali) [inline config/locales/devise.en.yml]: You don't need to include this file if you won't change its content

File: Gemfile (hunk 1 above)
Comment (bin0mial) [reply Gemfile]: Done c26ad2f822a81bf5e46f17b5d59eea99a8843fd2

File: config/locales/devise.en.yml (hunk 2 above)
Comment (bin0mial) [reply config/locales/devise.en.yml]: Done c26ad2f822a81bf5e46f17b5d59eea99a8843fd2

**Review by a-ali — APPROVED** [review]

**Pull request merged**